pkg/actionsmetrics/metrics.go

```diff
@@ -72,55 +72,56 @@ var (
 )
 
 var (
+	commonLabels                          = []string{"runs_on", "job_name", "organization", "repository", "workflow_name"}
 	githubWorkflowJobQueueDurationSeconds = prometheus.NewHistogramVec(
 		prometheus.HistogramOpts{
 			Name:    "github_workflow_job_queue_duration_seconds",
 			Help:    "Queue times for workflow jobs in seconds",
 			Buckets: runtimeBuckets,
 		},
-		[]string{"runs_on", "job_name"},
+		commonLabels,
 	)
 	githubWorkflowJobRunDurationSeconds = prometheus.NewHistogramVec(
 		prometheus.HistogramOpts{
 			Name:    "github_workflow_job_run_duration_seconds",
 			Help:    "Run times for workflow jobs in seconds",
 			Buckets: runtimeBuckets,
 		},
-		[]string{"runs_on", "job_name", "job_conclusion"},
+		append(commonLabels, "job_conclusion"),
```

Comment:
This isn't safe and can produce unexpected results for subsequent `append(commonLabels, ...)` calls, because append can mutate the slice.

Reply:
Ah interesting, sounds like that's more of an issue when using the colon slice notation but yeah seems sketchy!
Would you prefer to just use a literal and duplicate the common labels for each metric or try and retain the re-usable common label definition?

Reply:
I do like the idea of explicitly extracting and defining commonLabels!
How about adding a function to create a new copy of commonLabels, add some extra labels, and return the resulting slice? That way, we can repeat the function calls in places we currently refer to `commonLabels`.

Reply:
something lile:

```
func metricLabels(extras ...string) {
    return append(append([[]string{}, commonLabels...), extras...)
}
```

so that we can rewrite this line to
```suggestion
		metricLabels("job_conclusion"),
```